app/common/services/data-export.service.js

```diff
@@ -1,92 +1,68 @@
 module.exports = DataExport;
 
 
-DataExport.$inject = ['$rootScope', 'ConfigEndpoint', 'ExportJobEndpoint', '$q', '_', '$window', '$timeout', 'Notify', '$location'];
-function DataExport($rootScope, ConfigEndpoint, ExportJobEndpoint, $q, _, $window, $timeout, Notify, $location) {
-
-    function prepareExport(query) {
-        loadingStatus(true);
-        var site = ConfigEndpoint.get({ id: 'site' }).$promise;
-        var exportQuery = ExportJobEndpoint.save(query);
-
-        /**
-        TODOS:
-        1. Add polling to the api to check if the export-job is done,
-        i.e ExportJobEndpoint.get({id}) or similar, then notify the user
-        that their file is ready. 2. Keep in mind, below code will probably be changed depending on in what way the exported data is delivered
-        */
-
-        requestExport(site, query, exportQuery);
-    }
-
-    function requestExport(site, query, exportQuery) {
-        $q.all([site, exportQuery]).then(function (response) {
-            showCSVResults(response, query.format);
+DataExport.$inject = ['$rootScope', 'ExportJobEndpoint', 'Notify', '$window', '$timeout', '$interval', 'CONST'];
+function DataExport($rootScope, ExportJobEndpoint,  Notify, $window, $timeout, $interval, CONST) {
+    var polling;
+    function startExport(query) {
+        query.entity_type = 'post';
+        ExportJobEndpoint.save(query).$promise.then(function (job) {
+            loadingStatus(true, null, job);
+            polling = startPolling(job);
         }, function (err) {
             loadingStatus(false, err);
         });
     }
-    function cancelExport(jobId) {
-        //TODO: Add cancel-job
-        console.log('canceling job');
-    }
-
-    function showCSVResults(response, format) {
-        // Save export data to file
-        var filename = response[0].name + '-' + (new Date()).toISOString().substring(0, 10) + '.' + format,
-            data = response[1].data;
 
-        handleArrayBuffer(filename, data, 'csv');
-        loadingStatus(false);
-        $rootScope.$broadcast('event:data_export:complete');
-        return filename;
+    function startPolling(job) {
```

Comment:
This is where the actual polling starts. Have only been able to test it with mock-data.